dm/master/shardddl/optimist.go

```diff
@@ -291,6 +305,11 @@ func (o *Optimist) recoverLocks(
 					}
 					if op.Done {
 						lock.TryMarkDone(op.Source, op.UpSchema, op.UpTable)
+						err := lock.DeleteColumnsByDDLs(op.DDLs)
```

Comment:
hope we could left more comment, I think I might forget the logic somedays later

```
if DM-master sent a DROP COLUMN DDL, all shard tables had dropped that column and got synced. So we delete it from paritially dropped columns
```

Reply:
added under this function's definition.